Had to use latest msgspec (Python 3.12 to Python 3.13 upgrade)

Update from the SearXNG maintainers (20241229): if you have issues with the upgrade of the CPython (Python 3.12 to Python 3.13 upgrade), please read https://github.com/searxng/searxng/issues/4015#issuecomment-2564670601 first.

---

Using the latest commit (https://github.com/bojanstef/searxng/commit/abd9b271bc1c4cc4dc1ee531944ce27c1e2734cc) from my fork: https://github.com/bojanstef/searxng when I run the following command from the documentation here https://docs.searxng.org/admin/installation-docker.html#build-the-image
```shell
make docker.build
``` 


<details>
<summary>I get this failing output (click to expand)</summary>

```shell
Building wheels for collected packages: msgspec
  Building wheel for msgspec (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for msgspec (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [51 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build/lib.macosx-14.0-arm64-cpython-313/msgspec
      copying msgspec/msgpack.py -> build/lib.macosx-14.0-arm64-cpython-313/msgspec
      copying msgspec/_version.py -> build/lib.macosx-14.0-arm64-cpython-313/msgspec
      copying msgspec/__init__.py -> build/lib.macosx-14.0-arm64-cpython-313/msgspec
      copying msgspec/yaml.py -> build/lib.macosx-14.0-arm64-cpython-313/msgspec
      copying msgspec/structs.py -> build/lib.macosx-14.0-arm64-cpython-313/msgspec
      copying msgspec/inspect.py -> build/lib.macosx-14.0-arm64-cpython-313/msgspec
      copying msgspec/_json_schema.py -> build/lib.macosx-14.0-arm64-cpython-313/msgspec
      copying msgspec/json.py -> build/lib.macosx-14.0-arm64-cpython-313/msgspec
      copying msgspec/_utils.py -> build/lib.macosx-14.0-arm64-cpython-313/msgspec
      copying msgspec/toml.py -> build/lib.macosx-14.0-arm64-cpython-313/msgspec
      copying msgspec/msgpack.pyi -> build/lib.macosx-14.0-arm64-cpython-313/msgspec
      copying msgspec/__init__.pyi -> build/lib.macosx-14.0-arm64-cpython-313/msgspec
      copying msgspec/json.pyi -> build/lib.macosx-14.0-arm64-cpython-313/msgspec
      copying msgspec/structs.pyi -> build/lib.macosx-14.0-arm64-cpython-313/msgspec
      copying msgspec/py.typed -> build/lib.macosx-14.0-arm64-cpython-313/msgspec
      UPDATING build/lib.macosx-14.0-arm64-cpython-313/msgspec/_version.py
      set build/lib.macosx-14.0-arm64-cpython-313/msgspec/_version.py to '0.18.6'
      running build_ext
      building 'msgspec._core' extension
      creating build/temp.macosx-14.0-arm64-cpython-313/msgspec
      clang -fno-strict-overflow -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -O3 -Wall -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk -I/opt/homebrew/opt/openjdk@17/include -I/Users/bojanstef/Developer/searxng/local/py3/include -I/opt/homebrew/opt/python@3.13/Frameworks/Py
thon.framework/Versions/3.13/include/python3.13 -c msgspec/_core.c -o build/temp.macosx-14.0-arm64-cpython-313/msgspec/_core.o
      msgspec/_core.c:500:13: error: call to undeclared function '_PyUnicode_EQ'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
              if (_PyUnicode_EQ(kwname, key)) {
                  ^
      msgspec/_core.c:4452:12: error: call to undeclared function '_PySet_NextEntry'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
          while (_PySet_NextEntry(state->structs_set, &set_pos, &set_item, &set_hash)) {
                 ^
      msgspec/_core.c:7316:17: error: call to undeclared function '_PyUnicode_EQ'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
                  if (_PyUnicode_EQ(kwname, field)) {
                      ^
      msgspec/_core.c:7723:17: error: call to undeclared function '_PyUnicode_EQ'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
                  if (_PyUnicode_EQ(kwname, field)) goto kw_found;
                      ^
      msgspec/_core.c:11254:72: error: too few arguments to function call, expected 6, have 5
          int out = _PyLong_AsByteArray((PyLongObject *)int128, buf, 16, 0, 0);
                    ~~~~~~~~~~~~~~~~~~~                                      ^
      /opt/homebrew/opt/python@3.13/Frameworks/Python.framework/Versions/3.13/include/python3.13/cpython/longobject.h:111:17: note: '_PyLong_AsByteArray' declared here
      PyAPI_FUNC(int) _PyLong_AsByteArray(PyLongObject* v,
                      ^
      msgspec/_core.c:12426:12: error: call to undeclared function '_PySet_NextEntry'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
          while (_PySet_NextEntry(obj, &ppos, &item, &hash)) {
                 ^
      msgspec/_core.c:13748:12: error: call to undeclared function '_PySet_NextEntry'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
          while (_PySet_NextEntry(obj, &ppos, &item, &hash)) {
                 ^
      7 errors generated.
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for msgspec
Failed to build msgspec
ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (msgspec)
ERROR: ./manage: line 307: error while pip install (local/py3/bin)
make: *** [docker.build] Error 42
```

</details>

Searching, I found this related issue: https://github.com/jcrist/msgspec/issues/698#issuecomment-2428026677 and changing the requirements.txt file fixed the problem for me:
```diff
certifi==2024.8.30
babel==2.16.0
flask-babel==4.0.0
flask==3.0.3
jinja2==3.1.4
lxml==5.3.0
pygments==2.18.0
python-dateutil==2.9.0.post0
pyyaml==6.0.2
httpx[http2]==0.24.1
Brotli==1.1.0
uvloop==0.21.0
httpx-socks[asyncio]==0.7.7
setproctitle==1.3.3
redis==5.0.8
markdown-it-py==3.0.0
fasttext-predict==0.9.2.2
tomli==2.0.2; python_version < '3.11'
-msgspec==0.18.6
+msgspec @ git+https://github.com/jcrist/msgspec.git@main#egg=msgspec
eval_type_backport; python_version < '3.9'
typer-slim==0.13.0
```

--- a/.github/workflows/checker.yml
+++ b/.github/workflows/checker.yml
@@ -19,7 +19,7 @@ jobs:
       - name: Set up Python
         uses: actions/setup-python@v5
         with:
-          python-version: '3.11'
+          python-version: '3.13'
           architecture: 'x64'
 
       - name: Install Python dependencies

--- a/.github/workflows/integration.yml
+++ b/.github/workflows/integration.yml
@@ -16,7 +16,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-20.04]
-        python-version: ["3.9", "3.10", "3.11", "3.12"]
+        python-version: ["3.9", "3.10", "3.11", "3.12", "3.13"]
     steps:
       - name: Checkout
         uses: actions/checkout@v4

--- a/requirements.txt
+++ b/requirements.txt
@@ -17,6 +17,5 @@ markdown-it-py==3.0.0
 fasttext-predict==0.9.2.4
 tomli==2.0.2; python_version < '3.11'
 msgspec-python313-pre
-eval_type_backport; python_version < '3.9'
 typer-slim==0.15.1
 isodate==0.7.2

